### PR TITLE
Added Unit tests, Improved current line coverage

### LINKED ISSUE #12: New Unit tests

### PR DESCRIPTION
**Pull request by alexalap**
Unit tests added / expanded:
- Url.encode(), Url.decode() -> Correctly handling special characters and invalid encodings
- Url.queryLength()
- Url props interface
- Url.urlConfig()

Line coverage is currently above 80%

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
   },
   "github": "https://github.com/Mikhus/domurl",
   "devDependencies": {
+    "jest": "^28.1.0",
```
Comment (tbeni22) [inline package.json:43]: Why is this necessary? We are using mocha.

Comment (alexalap) [reply package.json:43]: It is unnecessary, it could be removed.

Comment (tbeni22) [reply package.json:43]: OK

**Review by tbeni22 — CHANGES_REQUESTED** [review]
Jest should be removed.

**Review by tbeni22 — APPROVED** [review]

**Review by tbeni22 — APPROVED** [review]
New tests contain a bit too much unrelated 'assert' statements.
Otherwise a good fix, and much better code coverage.
